lib/clients/websocket.js

```diff
@@ -1,111 +1,87 @@
-var signRequest = require('../../lib/request_signer').signRequest;
+const EventEmitter = require('events').EventEmitter;
+const Websocket = require('ws');
+const Utils = require('../utilities.js');
+const signRequest = require('../../lib/request_signer').signRequest;
 
 /**
  * Create a new connection to a websocket feed
  * @param productIDs {array} The GDAX products to listen to. Default: ['BTC-USD']
  * @param websocketURI {string} Optional. The websocker URL. Default: The official GDAX feed.
  * @param auth {object} Optional. An object containing your API ket details (key, secret & passphrase)
  */
-var WebsocketClient = function(productIDs, websocketURI, auth) {
-  var self = this;
-  self.productIDs = Utils.determineProductIDs(productIDs);
-  self.websocketURI = websocketURI || 'wss://ws-feed.gdax.com';
-  if (auth && !(auth.secret && auth.key && auth.passphrase)) {
-    throw new Error('Invalid or incomplete authentication credentials. You should either provide all of the secret, key and passphrase fields, or leave auth null');
-  }
-  self.auth = auth || {};
-  EventEmitter.call(self);
-  self.connect();
-};
-
-util.inherits(WebsocketClient, EventEmitter);
-
-_.assign(WebsocketClient.prototype, new function() {
-  var prototype = this;
-
-  prototype.connect = function() {
-    var self = this;
-
-    if (self.socket) {
-      self.socket.close();
+class WebsocketClient extends EventEmitter {
+  constructor(productIDs, websocketURI = 'wss://ws-feed.gdax.com', auth) {
+    super();
+    this.productIDs = Utils.determineProductIDs(productIDs);
+    this.websocketURI = websocketURI;
+    if (auth && !(auth.secret && auth.key && auth.passphrase)) {
+      throw new Error(
+        'Invalid or incomplete authentication credentials. You should either provide all of the secret, key and passphrase fields, or leave auth null'
+      );
     }
+    this.auth = auth || {};
+    this.connect();
+  }
 
-    self.socket = new Websocket(self.websocketURI);
-
-    self.socket.on('message', self.onMessage.bind(self));
-    self.socket.on('open', self.onOpen.bind(self));
-    self.socket.on('close', self.onClose.bind(self));
-    self.socket.on('error', self.onError.bind(self));
-  };
+  connect() {
+    if (this.socket) this.socket.close();
 
-  prototype.disconnect = function() {
-    var self = this;
+    this.socket = new Websocket(this.websocketURI);
 
-    if (!self.socket) {
-      throw "Could not disconnect (not connected)"
-    }
+    this.socket.on('message', this.onMessage.bind(this));
+    this.socket.on('open', this.onOpen.bind(this));
+    this.socket.on('close', this.onClose.bind(this));
+    this.socket.on('error', this.onError.bind(this));
+  }
 
-    self.socket.close();
-  };
+  disconnect() {
+    if (!this.socket) throw new Error('Could not disconnect (not connected)');
+    this.socket.close();
+  }
 
-  prototype.onOpen = function() {
-    var self = this;
-    self.emit('open');
+  onOpen() {
+    this.emit('open');
```

Comment:
Hmm, now that you point that out, I'm confused as well. This was the previous behavior. Thoughts @fb55?